ckanext/harvest/harvesters/base.py

```diff
@@ -288,9 +288,21 @@ def _create_or_update_package(self, package_dict, harvest_object,
 
             if self.config and self.config.get('clean_tags', False):
                 tags = package_dict.get('tags', [])
-                tags = [munge_tag(t) for t in tags if munge_tag(t) != '']
-                tags = list(set(tags))
-                package_dict['tags'] = tags
+                if tags:
+                    if not isinstance(tags[0], dict):
+                        # REST format: 'tags' is a list of strings
+                        tags = [munge_tag(t) for t in tags if munge_tag(t) != '']
+                        tags = list(set(tags))
+                        package_dict['tags'] = tags
+                    else:
+                        # package_show form: 'tags' is a list of dicts
+                        clean_tags = []
+                        for t in tags:
```

Comment:
Use a list comprehension just like above, no need for a `for` loop here.

Reply:
Uhm ok, but we're updating a field in a dict, guess I'll need something more complex